### PR TITLE
Allow deep linking

### PR DESCRIPTION
**Pull request by alex-ju**
### :pushpin: Summary

Allow deep linking – linking to headings or sections using URI fragments.

Preview link: https://hds-website-git-alex-ju-allow-fragment-navigation-hashicorp.vercel.app/components/alert#usage

### :hammer_and_wrench: Detailed description

There are two commits in this PR:
 - one that identifies when the fragment gets removed (there's a tradeoff here)
 - another one that proposes a solution for automatic scroll to heading/section, if the `id` exists on the page

Note: code is rather snappy, but wanted to get it up to get feedback

### :link: External links

Jira ticket: [HDS-1419](https://hashicorp.atlassian.net/browse/HDS-1419)

***

### 👀 Reviewer's checklist:

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-1419]: https://hashicorp.atlassian.net/browse/HDS-1419?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by MelSumner** [comment]
I see the delay in the scroll to ID. 

One of the tips I had for fixing this issue was to look at the implementation in the Ember deprecations app (https://github.com/ember-learn/deprecation-app) 

I wonder if you had a chance to test this out with a screen reader to see if we need to tweak the settings for `ember-a11y-refocus`?

File: website/app/controllers/show.js [hunk 1]
```diff
@@ -164,8 +164,6 @@ export default class ShowController extends Controller {
     if (this.tabs.length > 1 && this.currentActiveTabIndex != 0) {
       // for consistency we always set the query param tab to a lowercase version of the tab label
       set(this, 'selectedTab', this.tabs[current].label.toLowerCase());
-    } else {
-      set(this, 'selectedTab', null);
```
Comment (Dhaulagiri) [inline website/app/controllers/show.js:168]: this _was_ helpful to cleanup the URL is someone were to link to a query param that didn't exist, but in practice it does still work the same 🤔

Comment (alex-ju) [reply website/app/controllers/show.js:168]: yep, it still defaults to first tab, but doesn't clear the query param

**Comment by alex-ju** [comment]
> I see the delay in the scroll to ID.
> 
> One of the tips I had for fixing this issue was to look at the implementation in the Ember deprecations app (https://github.com/ember-learn/deprecation-app)
> 
> I wonder if you had a chance to test this out with a screen reader to see if we need to tweak the settings for `ember-a11y-refocus`?

Yes, tested with screen readers (bearing in mind the expected behavior we have using `ember-a11y-refocus`) and I don't think we need any adjustments there.

**Comment by alex-ju** [comment]
Will close this proof of concept in favor of #1167

**Pull request merged**